### PR TITLE
feat(llmclient): forward the client Idempotency-Key to providers

### PR DESCRIPTION
**Pull request by SantiagoDePolonia**
An `Idempotency-Key` sent by a client was dropped at the gateway. A provider could not recognize a replayed request, whether the client retried or GoModel's own retries did.

GoModel now forwards the client's key unchanged on the provider call it makes for the request, and on every llmclient retry of that call. The key is read from the request snapshot captured at ingress, so every route is covered without handler changes.

- **Failover attempts** go out without the key, because they send a different model's body under what would otherwise be the same key.
- **Gateway-internal calls** (guardrail and plugin origins) never carry it: they share the request context but send their own body.
- **An explicit header wins**, so passthrough behaves as before.
- **Validation:** keys must be visible ASCII of at most 255 characters; other values are ignored.

Verified against a fake upstream behind a two-target failover: the primary receives the client's key, the failover target receives none, and requests without a valid key send nothing.

### CODE REVIEW

File: internal/llmclient/idempotency_test.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	"github.com/enterpilot/gomodel/internal/core"
+)
+
+// keyRecordingServer answers failures first responses with 503, then 200, and
+// records the Idempotency-Key of every request it receives.
+func keyRecordingServer(t *testing.T, failures int) (*httptest.Server, func() []string) {
+	t.Helper()
+	var mu sync.Mutex
+	var keys []string
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```
[the remark this answers is not in the document]
Comment (SantiagoDePolonia) [reply internal/llmclient/idempotency_test.go:23]: This test cannot use `providertest`: it imports `internal/providers`, which imports `internal/llmclient`, so an `llmclient` test that imports it is an import cycle. The existing `llmclient` tests in `client_test.go` use `httptest` for the same reason. The fixture rule applies to provider packages, and this test exercises the HTTP client directly.

**Comment by SantiagoDePolonia** [comment]
On the review summary's point that invalid idempotency keys can reach the provider: I checked the validation against edge-case inputs on the committed branch. Values with a tab inside, DEL, CRLF, only whitespace, or longer than 255 bytes all forward nothing. A leading tab is trimmed. The one ambiguous case was a request that repeats the header: it used to forward the first value. As of e97a0f80, a repeated `Idempotency-Key` forwards nothing, covered by `TestIdempotencyKey_RepeatedHeaderForwardsNothing`. If the summary meant a different path, please point to it.
